docs/proposals/buildstrategy-steps-resources.md

```diff
@@ -0,0 +1,302 @@
+        - -group=/layers/group.toml
+        - -plan=/layers/plan.toml
+      volumeMounts:
+        - name: layers-dir
+          mountPath: /layers
+      resources:
+        limits:
+          cpu: "250m"
+          memory: "50Mi"
+        request:
+          cpu: "250m"
+          memory: "50Mi"
+    - name: step-restore
+      image: gcr.io/paketo-buildpacks/builder:latest
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/restorer
+      args:
+        - -layers=/layers
+        - -cache-dir=/cache
+        - -group=/layers/group.toml
+      volumeMounts:
+        - name: cache-dir
+          mountPath: /cache
+        - name: layers-dir
+          mountPath: /layers
+    - name: step-build
+      image: gcr.io/paketo-buildpacks/builder:latest
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/builder
+      args:
+        - -app=/workspace/source/$(build.source.contextDir)
+        - -layers=/layers
+        - -group=/layers/group.toml
+        - -plan=/layers/plan.toml
+      volumeMounts:
+        - name: layers-dir
+          mountPath: /layers
+      resources:
+        limits:
+          cpu: "500m"
+          memory: "1Gi"
+        request:
+          cpu: "500m"
+          memory: "1Gi"
+    - name: step-export
+      image: gcr.io/paketo-buildpacks/builder:latest
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/exporter
+      args:
+        - -app=/workspace/source/$(build.source.contextDir)
+        - -layers=/layers
+        - -cache-dir=/cache
+        - -group=/layers/group.toml
+        - $(build.output.image)
+      volumeMounts:
+        - name: cache-dir
+          mountPath: /cache
+        - name: layers-dir
+          mountPath: /layers
+```
+
+### Risks and Mitigations
+
+Proper documentation needs to be made to communicate the existance of `flavours` for strategies of the same type to users. When it comes to UI, is the decision of the UI team to decide
+how they will expose(_interface_) this flavours to their end-users. Strategy admins have full responsability on the flavours they decide to define, but recommendations of flavours should be
+made for this repository.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+#### Examples
+
+##### Dev Preview -> Tech Preview
+
+- N/A
+
+##### Tech Preview -> GA
+
+- N/A
+
+**For non-optional features moving to GA, the graduation criteria must include end to end tests.**
+
+##### Removing a deprecated feature
+
+- N/A
+
+### Upgrade / Downgrade Strategy
+
+- N/A
+
+### Version Skew Strategy
+
+- N/A
+
+## Implementation History
+
+- N/A
+
+## Drawbacks
+
+None. Only that we will remove a feature that is currently not performing properly.
+
+## Alternatives
+
+None at the moment.
```

Comment:
Even if it is not the path that is initially taken, the alternative I proposed in the community cabal meeting on Jun 29, where we create a separate CRD/struct for the resources settings and have the strategy have an optional reference to that should at least be noted.

Although "more involved", the pros include:
- incremental progress toward moving away from directly embedding corev1.Container in our types
- allow for separate/varying RBAC on the strategy objects vs. resource specification assigned to those strategies ... i.e. only certain strategies would be allowed to use more expensive resource allocations .... in case there are scenarios where devs can modify strategy objects, but only admins decide which resource allocations are assigned

@qu1queee @otaviof @sbose78

Reply:
sure, done, pls review again

Reply:
perfect / thanks @qu1queee !